File: subprojects/tools/src/registry.js
import { agentTools } from "#strap/tools/agent-tools";
import { fsTools } from "#strap/tools/fs-tools";
import { processTools } from "#strap/tools/process-tools";
import { jsmcpTools } from "#strap/jsmcp/tools";
import { scriptTools } from "#strap/tools/script-tools";
import { webTools } from "#strap/tools/web-tools";

export function getTools(group = "all") {
  const groups = {
    fs: fsTools(),
    process: processTools(),
    web: webTools(),
    agent: agentTools(),
    scripts: scriptTools(),
    jsmcp: jsmcpTools(),
  };
  if (group === "all") return Object.values(groups).flat();
  if (!groups[group]) throw new Error(`Unknown tool group: ${group}`);
  return groups[group];
}

export function toolMap(group = "all") {
  return new Map(getTools(group).map((tool) => [tool.name, tool]));
}

export function publicToolSpec(tool) {
  return {
    name: tool.name,
    description: tool.description,
    inputSchema: tool.inputSchema,
  };
}

File: subprojects/cli/commands/run-calls/src/run_calls/registry.clj
(ns run-calls.registry
  (:require [run-calls.agent-tools :as agent]
            [run-calls.fs-tools :as fs]
            [run-calls.jsmcp :as jsmcp]
            [run-calls.process-tools :as process]
            [run-calls.scripts :as scripts]
            [run-calls.web-tools :as web]))

(defn tools [group]
  (let [groups {"fs" fs/tools
                "process" process/tools
                "web" web/tools
                "agent" agent/tools
                "scripts" (scripts/tools)
                "jsmcp" jsmcp/tools}]
    (cond
      (= group "all") (apply merge (vals groups))
      (groups group) (groups group)
      :else (throw (ex-info (str "Unknown tool group: " group) {})))))
